test/controllers/CsvFileUploadControllerSpec.scala

```diff
@@ -695,69 +481,82 @@ class CsvFileUploadControllerSpec extends UnitSpec with GuiceOneAppPerSuite with
     }
 
     "redirect to validationFailure if validating fails" in {
       reset(mockErsConnector)
       when(
-        mockErsConnector.validateCsvFileData(any[List[CallbackData]](), any[SchemeInfo]())(any(), any(), any())
+        mockErsConnector.validateCsvFileData(any[List[UploadedSuccessfully]](), any[SchemeInfo]())(any(), any(), any())
       ).thenReturn(
         Future.successful(HttpResponse(ACCEPTED))
       )
 
-      val result = await(csvFileUploadController.validateCsv(mock[List[CallbackData]], mock[SchemeInfo])(Fixtures.buildFakeUser, Fixtures.buildFakeRequestWithSessionIdCSOP("GET"), hc))
+      val result = await(csvFileUploadController.validateCsv(mock[List[UploadedSuccessfully]], mock[SchemeInfo])(Fixtures.buildFakeUser, Fixtures.buildFakeRequestWithSessionIdCSOP("GET"), hc))
       status(result) shouldBe SEE_OTHER
       result.header.headers("Location") shouldBe routes.CsvFileUploadController.validationFailure().toString
     }
 
     "show error page if validating returns result other than OK and ACCEPTED" in {
       reset(mockErsConnector)
       when(
-        mockErsConnector.validateCsvFileData(any[List[CallbackData]](), any[SchemeInfo]())(any(), any(), any())
+        mockErsConnector.validateCsvFileData(any[List[UploadedSuccessfully]](), any[SchemeInfo]())(any(), any(), any())
       ).thenReturn(
         Future.successful(HttpResponse(INTERNAL_SERVER_ERROR))
       )
 
-      val result = await(csvFileUploadController.validateCsv(mock[List[CallbackData]], mock[SchemeInfo])(Fixtures.buildFakeUser, Fixtures.buildFakeRequestWithSessionIdCSOP("GET"), hc))
+      val result = await(csvFileUploadController.validateCsv(mock[List[UploadedSuccessfully]], mock[SchemeInfo])(Fixtures.buildFakeUser, Fixtures.buildFakeRequestWithSessionIdCSOP("GET"), hc))
       status(result) shouldBe OK
     }
 
     "direct to ers errors page if connecting with validator is not successful" in {
       reset(mockErsConnector)
       when(
-        mockErsConnector.validateCsvFileData(any[List[CallbackData]](), any[SchemeInfo]())(any(), any(), any())
+        mockErsConnector.validateCsvFileData(any[List[UploadedSuccessfully]](), any[SchemeInfo]())(any(), any(), any())
       ).thenReturn(
         Future.failed(new RuntimeException)
       )
-
-      contentAsString(await(csvFileUploadController.validateCsv(mock[List[CallbackData]], mock[SchemeInfo])(Fixtures.buildFakeUser, Fixtures.buildFakeRequestWithSessionIdCSOP("GET"), hc))) shouldBe contentAsString(csvFileUploadController.getGlobalErrorPage)
+      contentAsString(await(csvFileUploadController.validateCsv(mock[List[UploadedSuccessfully]], mock[SchemeInfo])(Fixtures.buildFakeUser, Fixtures.buildFakeRequestWithSessionIdCSOP("GET"), hc))) shouldBe contentAsString(csvFileUploadController.getGlobalErrorPage)
     }
+  }
+
+  override def fakeApplication(): Application = new GuiceApplicationBuilder().configure(config).build()
```

Comment:
if we are using this can we just override the bindings for our mocks

Reply:
Controllers are DI'd, so no.

Reply:
Please dont ask me to DI controllers.

Reply:
doh